CONTRIBUTING.md

```diff
@@ -0,0 +1,141 @@
+- **Make sure the appropriate licence header appears at the top of your source code file.** We use [LGPL v2.1](http://opensource.org/licenses/LGPL-2.1 "The GNU Lesser General Public License, version 2.1") for eXist and *strongly* encourage that, but ultimately any compatible [OSI approved license](http://opensource.org/licenses "Open Source Licenses") without further restrictions may be used.
+- **Run the full eXist test suite.** We don't accept code that causes regressions.
+
+
+Do I work on a bug-fix using a `feature` or a `hotfix`?
+-------------------------------------------------------
+If you want to contribute a *bug-fix*, you need to consider whether this is a `feature` or a `hotfix` in GitFlow terminology.
+
+Making the determination involves considering how the bug-fix is to be applied and how it is to be applied. First, you should carefully read the "Feature branches" and "Hotfix branches" sections from [A successful Git branching model](http://nvie.com/posts/a-successful-git-branching-model/). 
+
+If you're still unsure, consider: 
+
+- The bug-fix is a hotfix if it is **critical** and needs to go into a very soon to be released revision version i.e. 2.1.n to address an immediate production issue.
+- Otherwise it is a feature, i.e. its just standard development towards the next release of eXist.
+
+Even for a bug-fix you should most probably use a `feature`. If you're certain you want to create a `hotfix`, please consider discussing first via the `exist-development` mailing list.
+
+
+Help! I am a human, what does this all mean?
+--------------------------------------------
+- You work in features using GitFlow in your own fork of our repo.
+- If you want to push your feature to your fork before you have finished it locally, i.e. for the purposes of backup or collaboration, you can use `git flow feature publish my-feature`.
+- You will only ever send Pull Requests between your 'develop' branch and our 'develop' branch. i.e. finished features.
+- If you follow the details above and make it easy for us to accept your Pull Requests, they will get accepted and merged quickly!
+
+
+Syncing a Fork
+--------------
+Your fork will eventually become out of sync with the upstream repo as others contribute to eXist. To pull upstream changes into your fork, you have two options:
+
+1. [Merging](https://help.github.com/articles/syncing-a-fork).
+2. Rebasing.
+
+Rebasing leads to a cleaner revision history which is much easier to follow and is our preferred approach. However, `git rebase` is a very sharp tool and must be used with care. For those new to rebase, we would suggest having a backup of your local (and possibly remote) git repos before continuing. Read on to learn how to sync using rebase.
```

Comment:
I did not know. But this text was copied as-is from the original README.md ; if it is not the best text/procedure we need to update it, preferable in a new PR.